sqlglot/optimizer/merge_subqueries.py

```diff
@@ -242,6 +243,7 @@ def _merge_from(outer_scope, inner_scope, node_to_replace, alias):
         alias (str)
     """
     new_subquery = inner_scope.expression.args["from"].this
+    new_subquery.set("joins", node_to_replace.args.get("joins"))
```

Comment:
This was added because we can have `<CTEs> SELECT ... FROM (tbl1 JOIN tbl2)`. If we just merged a CTE into `tbl1`, then we'd lose the `JOIN tbl2` part because it's attached on the Table.

A concrete example (without this addition):

```python
>>> from sqlglot.optimizer import optimize
>>> schema = {"x": {"a": "int", "b": "int"}, "y": {"b": "int", "c": "int"}}
>>> optimize("SELECT b FROM ((SELECT a FROM x) INNER JOIN y ON a = b)", schema=schema).sql()
'SELECT "y"."b" AS "b" FROM ("x" AS "x")'
```

The expression just before merging CTEs is:

```sql
WITH _q_0 AS (SELECT x.a AS a FROM x AS x), y_2 AS (SELECT y.b AS b FROM y AS y) SELECT y.b AS b FROM (_q_0 AS _q_0 JOIN y_2 AS y ON _q_0.a = y.b)
```

We basically replace `_q_0` with the first CTEs table name and we lose the join, as explained above.